### ISSUE TITLE
Add 24-hour clock option to the UI

### ISSUE THREAD

**Issue by kevingrismore** [PrefectHQ/prefect#15078]
### Describe the current behavior

A user cannot customize the current 12-hour clock displays for times in the UI.

### Describe the proposed behavior

A user should be able to customize their UI preferences such that a 24-hour clock is selectable for all time displays across the UI. This is preferable for standardizing time displays, alongside the existing time zone customization, for globally distributed teams.

**Comment by pleek91** [comment]
It would be nice to let users customize the timezone and the format ❤️

**Comment by OrjanPettersen** [comment]
I agree.

**Comment by grahamstratton** [comment]
Yes please. Or make 24 hour times the only option! AM/PM is so much easier to mis-read.
